docs/publish.md

```diff
@@ -2661,6 +2662,123 @@ Here's an example of how it will look on Android:
+
+Here's an example showing how to send custom fields:
+
+=== "Command line (curl) (JSON)"
+    ```
+    curl ntfy.sh \
+      -d '{
+        "topic": "mytopic",
+        "message": "Disk space is low at 5.1 GB",
+        "title": "Low disk space alert",
+        "tags": ["warning","cd"],
+        "priority": 4,
+        "extras": {"lastChecked": "20230205"}
+      }'
+    ```
+=== "Command line (curl) (Header)"
+    ```
+    curl \
+        -H "Title: Low disk space alert" \
+        -H "Tags: warning,cd" \
+        -H "X-Priority: 4" \
+        -H 'X-Extras: {"lastChecked": "20230205"}' \
+        -d "Disk space is low at 5.1 GB" \
+        ntfy.sh/mytopic
+    ```
+
+=== "HTTP"
+    ``` http
+    POST /mytopic HTTP/1.1
+    Host: ntfy.sh
+    Title: Low disk space alert
+    Tags: warning,cd
+    X-Priority: 4
+    X-Extras: {"lastChecked": "20230205"}
+
+    Disk space is low at 5.1 GB
+    ```
+
+=== "JavaScript"
+    ``` javascript
+    fetch('https://ntfy.sh/mytopic', {
+        method: 'POST',
+        headers: {
+            'Title': 'Low disk space alert',
+            'Tags': 'warning,cd'
+            'X-Priority': '4',
+            'X-Extras': {'lastChecked': '20230205'}
+        },
+        body: "Disk space is low at 5.1 GB"
+    })
+    ```
+
+=== "Go"
+    ``` go
+    req, _ := http.NewRequest("POST", "https://ntfy.sh/mytopic", strings.NewReader("Disk space is low at 5.1 GB"))
+    req.Header.Set("Title", "Low disk space alert")
+    req.Header.Set("Tags", "warning,cd")
+    req.Header.Set("X-Priority", "4")
+    req.Header.Set("X-Extras", `{"lastChecked": "20230205"}`)
+    http.DefaultClient.Do(req)
+    ```
+
+=== "PowerShell"
+    ``` powershell
+    $Request = @{
+      Method = "POST"
+      URI = "https://ntfy.sh"
+      Body = @{
+        Topic    = "mytopic"
+        Title    = "Low disk space alert"
+        Tags     = @("warning", "cd")
+        Priority = 4
+        Message  = "Disk space is low at 5.1 GB"
+        Extras   = ConvertTo-JSON @{
+          lastChecked = "20230205"
+        }
+      }
+      ContentType = "application/json"
+    }
+    Invoke-RestMethod @Request
+    ```
+
+=== "Python"
+    ``` python
+    requests.post("https://ntfy.sh/mytopic",
+        data="Disk space is low at 5.1 GB",
+        headers={
+            "Title": "Low disk space alert",
+            "Tags": "warning,cd",
+            "X-Priority": "4",
+            "X-Extras": '{"lastChecked": "20230205"}'
+        })
+    ```
+
+=== "PHP"
+    ``` php-inline
+    file_get_contents('https://ntfy.sh/mytopic', false, stream_context_create([
+        'http' => [
+        'method' => 'PUT',
+        'header' =>
+            "Content-Type: text/plain\r\n" . // Does not matter
+            "Title: Low disk space alert\r\n" .
+            "Tags: warning,cd\r\n" .
+            "X-Priority: 4\r\n" .
+            "X-Extras: {\"lastChecked\": \"20230205\"}",
```

Comment:
Welllllll, that is just really shitty syntax, no? I feel like we should also add the "simple" syntax we have for the actions here. We could probably re-use the parser for most of it, though it's suddenly a lot harder.

Reply:
I didn't test most of these and don't know if the syntax is actually correct or if there's an easier way to write it out, especially with this PHP one. I'll look into adding the "simple" syntax, though